### ISSUE TITLE
go get x509: failed to load system roots and no roots provided

### ISSUE THREAD

**Issue by andrewrynhard** [NixOS/nixpkgs#10875]
Installed go with nix in a nix-shell environment and I am getting errors when using `go get`. As the title explains the error is `x509: failed to load system roots and no roots provided`. I have made sure that the `nix.sh` environment variables are set but still get this error. This is on OS X.

**Comment by pikajude** [comment]
go being unable to interoperate with the system keychain is a known issue (`hub` is also broken), though it needs more investigation.

**Comment by nbp** [comment]
cc @cstrahan @wkennington

**Comment by andrewrynhard** [comment]
@pikajude this with go1.5.

**Comment by zimbatm** [comment]
The nix profile sets the SSL_CERT_FILE environment variable but go doesn't seem to take that into account.

**Comment by chrisfarms** [comment]
There's an [open Go issue](https://github.com/golang/go/issues/14311) to support `SSL_CERT_FILE` which should resolve this.

In the meantime I think you need to ensure `/etc/ssl/certs/ca-bundle.crt` exists, since Go appears to happily guess that location.

this did the trick for me:

```
ln -s $SSL_CERT_FILE /etc/ssl/certs/ca-bundle.crt
```

**Comment by zimbatm** [comment]
I actually have a patch to add SSL_CERT_FILE support to go but then decided it was too much work maintaining the patch. If you're interested it's at https://github.com/zimbatm/nixpkgs/commit/e5557a69d808f6532ef541510f4fdd4f6b380003 . Both go 1.4 and 1.5 are supported. I did it before go 1.6 so that would need to be adapted.

**Comment by ben0x539** [comment]
yo I didn't get very far trying to figure this out. The thing with putting the certs into `/etc` didn't help for me. I've been using the following test case:

```go
package main
import ( "fmt"; "net/http")
func main() { fmt.Println(http.Get("https://nixos.org/channels/nixos-unstable/git-revision")) }
```

It fails with the error message in the issue title.

Trying to dtrace (I'm not very good with OS X) it, I noticed that it works when running as root. Unclear what the deal is, looks like there's some certificate data scrolling past in the trace either way.

Does go1.7 change the behavior? I haven't been able to get it to build so far.

**Comment by mstone** [comment]
@andrewrynhard does https://github.com/mstone/nixpkgs/commit/fc288790da4191225a80db2be9d3c229091c859f work around the problem for you?

**Comment by LnL7** [comment]
This is fixed for 1.7
